### PR TITLE
Redux and Routing tasks

### PR DESCRIPTION
**Pull request by mamayserg**

### CODE REVIEW

File: src/components/filterTab/FilterTab.js [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
   return (
     <div className="text-center text-xs	text-white font-light border-b border-gray-200  flex justify-between ">
       <ul className="flex flex-wrap -mb-px">
         <li className="mr-2">
           {tabs.map((item, i) => (
             <TabItem
-              key={item.name}
-              path={item.path}
-              active={location.hash === item.path.hash}
+              key={i}
```
Comment (ambroseus) [inline src/components/filterTab/FilterTab.js:24]: good practice is not to use index but use some uniq item prop (id, hash, etc)

File: src/components/movieItem/MovieItem.js [hunk 2]
```diff
@@ -2,27 +2,35 @@ import MovieGenre from "./MovieGenre";
...
-      <div className="flex flex-row align-top justify-between mt-3">
-        <div className="flex flex-col">
-          <MovieName name={movie.title} />
-          <MovieGenre genre={movie.genre} />
-        </div>
-        <div>
-          <MovieYear year={movie.year} />
-        </div>
-      </div>
+  return (<>{isLoading ? (<div className="mb-10">
```
Comment (ambroseus) [inline src/components/movieItem/MovieItem.js:10]: personally, I'd prefer to use a more readable approach:
```
<>
  {isLoading && (.......)}
  {!isLoading && (......)}
</>

File: src/components/selectedMovieCard/SelectedMovieCard.js [hunk 3]
```diff
@@ -1,22 +1,112 @@
...
+  let { id } = useParams();
+
+  const { data, isLoading, error } = useGetMovieQuery(id);
+
+  useEffect(() => {
+    if (error) {
+      navigate(`/search`);
+    }
+    // eslint-disable-next-line
+  }, error);
```
Comment (ambroseus) [inline src/components/selectedMovieCard/SelectedMovieCard.js:23]: Is error an array? Hook's dependencies must always be arrays

**Review by ambroseus — APPROVED** [review]

File: src/components/filterTab/FilterTab.js (hunk 1 above)
Comment (mamayserg) [reply src/components/filterTab/FilterTab.js:24]: @ambroseus  in this case, this array is not dynamic ( I understand your concern )

**Pull request merged**